File: 1319-unique-number-of-occurrences/1319-unique-number-of-occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        HashMap<Integer,Integer> Map=new HashMap<>();
        Set<Integer> out=new HashSet<>();
    
        for(int i=0;i<arr.length;i++)
        {
           Map.put(arr[i],Map.getOrDefault(arr[i],0)+1);
        }
        for(int num:Map.values())
        {
            out.add(num);
        }
        
        return Map.size()!=out.size() ? false :true;

    }
}

File: 1319-unique-number-of-occurrences/1319-unique-number-of-occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:

        frequency_map = {}

        for num in arr:
            frequency_map[num] = frequency_map.get(num,0)+1
        
        res_set = set()

        for key,value in frequency_map.items():
            if value in res_set:
                return False
            else :
                res_set.add(value)
        

        return True
        